### PR TITLE
tv-casting-app: Fixing caching logic and app crashes

### LINKED ISSUE #25377: [BUG] examples/tv-casting-app: Unable to match with previously connected TV if deviceName changes
### Reproduction steps

1. Commission the tv-casting-app with the tv-app
2. Kill the apps
3. Change the deviceName of the tv-app by setting CHIP_DEVICE_CONFIG_DEVICE_NAME and rebuild
4. Run the tv-casting-app and the tv-app.

Expected: The tv-casting-app starts casting (sends the content launcher command) to the tv-app on selecting it from the list of discovered commissioners
Actual: The tv-casting-app goes through commissioning again with the tv-app, before casting.

### Bug prevalence

Every time

### GitHub hash of the SDK that was being used

7ce9d7886c167790990159c659bf94b39af81976

### Platform

other

### Platform Version(s)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sharadb-amazon**
Fixes https://github.com/project-chip/connectedhomeip/issues/25377 and https://github.com/project-chip/connectedhomeip/issues/25379

### Change summary
1. Caching key change (3bd389f3f2e3300bbf48cfc6e540041cf1139a8a) - The first commit here changes the caching logic to prioritize hostnames to do the matching with previously discovered TV records.
2. The second commit (f869bc4723b686e606dcc4e3881135e672cf3ccf) changes the way we pass in the context to the FindOrEstablishSession API and when mActiveTargetVideoPlayerInfo was set in the CastingServer::verifyOrEstablishConnection() API. That fixes crashes we were observing earlier.

### Testing
Tested with the Android/iOS tv-casting-apps against the Linux tv-app

### CODE REVIEW

File: examples/tv-casting-app/tv-casting-common/src/TargetVideoPlayerInfo.cpp [hunk 1]
```diff
@@ -60,11 +71,14 @@ CHIP_ERROR TargetVideoPlayerInfo::Initialize(NodeId nodeId, FabricIndex fabricIn
...
                                                              std::function<void(CHIP_ERROR)> onConnectionFailure)
 {
-    mOnConnectionSuccessClientCallback = onConnectionSuccess;
-    mOnConnectionFailureClientCallback = onConnectionFailure;
-    Server * server                    = &(chip::Server::GetInstance());
-    server->GetCASESessionManager()->FindOrEstablishSession(ScopedNodeId(mNodeId, mFabricIndex), &mOnConnectedCallback,
-                                                            &mOnConnectionFailureCallback);
+    ChipLogProgress(AppServer, "TargetVideoPlayerInfo::FindOrEstablishCASESession called");
+
+    VideoPlayerConnectionContext * connectionContext = new VideoPlayerConnectionContext(
```
Comment (bzbarsky-apple) [inline examples/tv-casting-app/tv-casting-common/src/TargetVideoPlayerInfo.cpp:76]: Where is this VideoPlayerConnectionContext destroyed?

Comment (sharadb-amazon) [reply examples/tv-casting-app/tv-casting-common/src/TargetVideoPlayerInfo.cpp:76]: See line #136 and #161 in examples/tv-casting-app/tv-casting-common/include/TargetVideoPlayerInfo.h

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/tv-casting-app/darwin/TvCasting/TvCasting.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -428,7 +428,7 @@
 					CoreData,
 					"-Wl,-unexported_symbol,\"__Z*\"",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = com.matter.TvCasting;
+				PRODUCT_BUNDLE_IDENTIFIER = "com.matter.TvCasting-sharadb";
```
Comment (chrisdecenzo) [inline examples/tv-casting-app/darwin/TvCasting/TvCasting.xcodeproj/project.pbxproj:431]: was this intentional?

File: examples/tv-casting-app/darwin/TvCasting/TvCasting.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -475,7 +475,7 @@
 					"-Wformat-nonliteral",
 					"-Wformat-security",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = com.matter.TvCasting;
+				PRODUCT_BUNDLE_IDENTIFIER = "com.matter.TvCasting-sharadb";
```
Comment (chrisdecenzo) [inline examples/tv-casting-app/darwin/TvCasting/TvCasting.xcodeproj/project.pbxproj:478]: intentional?

**Review by chrisdecenzo — APPROVED** [review]

File: examples/tv-casting-app/darwin/TvCasting/TvCasting.xcodeproj/project.pbxproj (hunk 2 above)
Comment (sharadb-amazon) [reply examples/tv-casting-app/darwin/TvCasting/TvCasting.xcodeproj/project.pbxproj]: No. Removed this diff.

File: examples/tv-casting-app/darwin/TvCasting/TvCasting.xcodeproj/project.pbxproj (hunk 3 above)
Comment (sharadb-amazon) [reply examples/tv-casting-app/darwin/TvCasting/TvCasting.xcodeproj/project.pbxproj]: No. Removed this diff.

**Review by cliffamzn — APPROVED** [review]

**Pull request merged**